scripts/i18n.js

```diff
@@ -51,18 +52,23 @@ export const loadTranslations = async () => {
 
     const locale = fname.slice(0, -3);
     const pofile = path.join(LOCALES_DIR, fname);
-    const po = PO.parse(await fs.readFile(pofile, 'utf8'));
+    // eslint-disable-next-line no-await-in-loop
+    const pofileContent = await fs.readFile(pofile, 'utf8');
+    const po = PO.parse(pofileContent);
     const translations = {};
     for (const item of po.items) {
       translations[item.msgid] = item.msgstr[0];
     }
+
     locales[locale] = translations;
   }
 
   const defaultTranslations = {};
   for (const msgid in locales[Object.keys(locales)[0]]) {
+    if (!msgid) continue;
```

Comment:
Empty msgids are not valid PO entries, why do you check them here?

Reply:
https://eslint.org/docs/latest/rules/guard-for-in

You have no idea what the `locales` is. The code between `const locales = {}` and `locales[locale] = translations` may bugged. So the the `locales` is possibly `{}`.

Reply:
I just refactored the code with:

```js
const defaultTranslations = Object.keys(Object.values(locales).at(0)).map(
  (msgid) => ({
    [msgid]: msgid,
  }),
);
```

It will throw an error if the data is incorrect.

Reply:
That is not the same. You're generating a `Array<Object<String, String>>`, but the others locales are defined as `Object<String, String>`. The code works because the `|| msgid` in the `i18n()` function.

Just apply the next diff, please:

```diff
-  const defaultTranslations = Object.keys(Object.values(locales).at(0)).map(
-    (msgid) => ({
-      [msgid]: msgid,
-    }),
-  );
-
-  locales.en = defaultTranslations;
+  locales.en = {};
```